test/test-ResourceParser.cc

```diff
@@ -1009,3 +1009,32 @@ TEST_CASE("Relation identifiers should be unique for a resource", "[resource]")
     REQUIRE(resource.node.actions[0].relation.str == "create");
     REQUIRE(resource.node.actions[1].relation.str == "create");
 }
+
+TEST_CASE("Detect invalid reference to URI Tempalte parameters", "[resource]")
+{
+    mdp::ByteBuffer source = \
+    "## Orders [/orders]\n\n"\
```

Comment:
So this part is a different issue, it doesn't even bother to check the parameters at all. So in this case you don't get a warning at all.